### PR TITLE
Add calibration script

### PR DESCRIPTION
**Pull request by aegger13**
## Description

Add the calibration script

## Motivation and Context

calibrate jet tracking runs

## How Has This Been Tested?

Run on calibration data from Fromme
python jt_cal.py --exp cxilv4718 --run 172

have to source /reg/g/psdm/sw/conda1/manage/bin/psconda.sh

### CODE REVIEW

File: jet_tracking/jet_tracking_cal/submit_jt_arp.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+source /reg/g/psdm/sw/conda1/manage/bin/psconda.sh
+ABS_PATH=/cds/home/a/aegger/jet_tracking/jet_tracking/jet_tracking_cal
```
Comment (ZryletTC) [inline jet_tracking/jet_tracking_cal/submit_jt_arp.sh:4]:
```suggestion
ABS_PATH=`dirname $(readlink -f $0)`
```

File: jet_tracking/jet_tracking_cal/submit_jt_arp.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+sbatch --nodes=2 --time=5 $ABS_PATH/jt_cal.py "$@"
```
Comment (ZryletTC) [inline jet_tracking/jet_tracking_cal/submit_jt_arp.sh:5]: Where is `sbatch` meant to come from? I don't see in on my path after sourcing `/reg/g/psdm/sw/conda1/manage/bin/psconda.sh`

File: jet_tracking/jet_tracking_cal/jt_cal.py [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+    parser.add_argument('--bins', type=int, \
+        default=(BINS))
+    args = parser.parse_args()
+
+    # All useful information should come from the config file
+    # This will be source of truth for starting point for each hutch for now
+    with open(args.cfg) as f:
+        yml_dict = yaml.load(f, Loader=yaml.FullLoader)
+        det_map = yml_dict['det_map']
+        ipm_name = yml_dict['ipm']['name']
```
Comment (ZryletTC) [inline jet_tracking/jet_tracking_cal/jt_cal.py:346]: Can we swap `ipm` for `i0` since it won't always be an ipm?

File: jet_tracking/__init__.py [hunk 4]
```diff
@@ -1,10 +0,0 @@
-from . import cam_utils  # noqa
```
Comment (ZryletTC) [inline jet_tracking/__init__.py:1]: Why'd you remove this file?

File: jet_tracking/jet_tracking_cal/jt_cal.py [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+from bokeh.models import Span, Legend, LegendItem, ColorBar, LinearColorMapper
+from bokeh.io import output_notebook
+import panel as pn
+from scipy.stats import gaussian_kde
+
+logger = logging.getLogger(__name__)
+
+# Constants
+# Location of my branch for testing
+JT_LOC = '/cds/home/a/aegger/jet_tracking/jet_tracking/'
```
Comment (ZryletTC) [inline jet_tracking/jet_tracking_cal/jt_cal.py:24]: Should be unhardcoded

File: jet_tracking/jet_tracking_cal/jt_cal.py [hunk 6]
```diff
@@ -0,0 +1,441 @@
...
+    with open(args.cfg) as f:
+        yml_dict = yaml.load(f, Loader=yaml.FullLoader)
+        det_map = yml_dict['det_map']
+        ipm_name = yml_dict['ipm']['name']
+        ipm_det = yml_dict['ipm']['det']
+        hutch = yml_dict['hutch']
+
+    # Setup data source and write directories
+    exp_run = ''.join(['exp=', args.exp, ':run=', args.run])
+    base_dir = ''.join(['/cds/data/psdm/', hutch, '/', args.exp, '/'])
```
Comment (ZryletTC) [inline jet_tracking/jet_tracking_cal/jt_cal.py:352]: While I get these folders likely won't change in our filesystem a long while, can they be moved to the yaml as it seems like a configuration value and just for portability's sake?

File: jet_tracking/jet_tracking_cal/jt_cal.py [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+    return fig
+
+if __name__ == '__main__':
+    parser = ArgumentParser()
+    parser.add_argument('--exp', type=str, \
+        default=(os.environ.get('EXPERIMENT', None)))
+    parser.add_argument('--run', type=str, \
+        default=(os.environ.get('RUN_NUM', None)))
+    parser.add_argument('--cfg', type=str, \
+        default=(''.join([JT_LOC, 'mpi_scripts/mpi_configs/default_config.yml'])))
```
Comment (ZryletTC) [inline jet_tracking/jet_tracking_cal/jt_cal.py:334]: Good for now. Todo for later: add some autodetection, sort of like happi does for its cfg

File: jet_tracking/mpi_scripts/mpi_configs/default_config.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+api_msg:
```
Comment (ZryletTC) [inline jet_tracking/mpi_scripts/mpi_configs/default_config.yml:1]: Could this file be moved up to higher level, as it doesn't seem to be specific to mpi? Seems kinda buried at the moment esp with nothing else in `mpi_scripts`

File: jet_tracking/mpi_scripts/mpi_configs/default_config.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  name: 'jungfrau4M'
+  shape:
+    - 2203
+    - 2299
+  dtype: float32
+  bins: 100
+
+pv_map:
+  1: 'CXI:JTRK:REQ:DIFF_INTENSITY'
+  2: 'CXI:JTRK:REQ:I0'
```
Comment (ZryletTC) [inline jet_tracking/mpi_scripts/mpi_configs/default_config.yml:14]: Are these for other mpi_scripts?

**Review by ZryletTC — COMMENTED** [review]
Haven't looked at the nitty-gritty of the calibration code yet. Just some preliminary, mostly organizational, notes for now.

I'd rather not include the jupyter notebook in here. The jt_cal.py script should emulate it pretty well, so we don't need both, right?

I'm wondering whether we need a `jet_tracking_cal` folder. What if we moved jt_cal up a level and the arp script to a bin folder in the top level?

File: jet_tracking/jet_tracking_cal/submit_jt_arp.sh (hunk 2 above)
Comment (aegger13) [reply jet_tracking/jet_tracking_cal/submit_jt_arp.sh:5]: sbatch is recognized on psana for slurm batch jobs.

File: jet_tracking/__init__.py (hunk 4 above)
Comment (ZryletTC) [reply jet_tracking/__init__.py]: Yup #20 scrapped old stuff

**Comment by aegger13** [comment]
Just a note, cleanup is coming, a lot of little things to fix here...  Sorry for the hard coding and magic numbers, I think I know exactly how this should work now.

**Comment by ZryletTC** [comment]
@amandashack @vespos So is this ready to merge now? Further changes can be addressed in another PR

**Comment by vespos** [comment]
I think it is. There are still improvements needed, but I thing it is at a good place to be merged for now.

**Review by ZryletTC — APPROVED** [review]
Haven't rereviewed but approving for the merge. Trusting Adam and the tests that its good enough.

**Pull request merged**